src/test/CharterManager-unit.t.sol

```diff
@@ -403,6 +410,61 @@ contract CharterManagerTest is TestBase {
         assertEq(a.gems(), 100 * 1e18);
     }
 
+    function test_migration() public {
+        (Usr old,) = init_user();
+        (Usr a,) = init_user();
+
+        // setup old (different) ilk with the same gem to migrate out from
+        bytes32 oldIlk = "TOKEN-B";
+        vat.init(oldIlk);
+        vat.file(oldIlk, "line", CEILING * 1e27);
+        vat.file(oldIlk, "spot", 1e27);
+
+        GemJoin5 oldAdapter = new GemJoin5(address(vat), oldIlk, address(gem));
+        vat.rely(address(oldAdapter));
+
+        // setup old vault
+        old.approve(address(gem), address(oldAdapter));
+        old.joinDirect(address(oldAdapter), 60 * 1e6);
+        old.frobDirect(oldIlk, address(old), address(old), address(old), 60 * 1e18, 40 * 1e18);
+
+        (uint256 ink, uint256 art) = vat.urns(oldIlk, address(old));
+        assertEq(ink, 60 * 1e18);
+        assertEq(art, 40 * 1e18);
+        assertEq(old.dai(), 40 * 1e45);
+        assertEq(old.gems(), 0);
+
+        // set different rates per ilk
+        vat.fold(oldIlk, address(vow), 0.05 * 1e27); // old rate is 1.05
+        vat.fold(ilk,    address(vow), 0.02 * 1e27); // new rate is 1.02
+
+        // perform privileged migration sequence
+        (, uint256 oldRate,,,) = vat.ilks(oldIlk);
+        (, uint256 rate,,,) = vat.ilks(ilk);
+
+        int256  dart = int256(mul(oldRate, art) / rate);
+        int256  dink = int256(ink);
+        uint256 dec = Token(gem).decimals();
+        uint256 gemAmt = ink / (10 ** (18 - dec));
+
+        vat.grab(oldIlk, address(old), address(this), address(this), -int256(ink), -int256(art));
+        oldAdapter.exit(address(this), gemAmt);
+        Token(gem).approve(address(manager), gemAmt);
+        manager.join(address(adapter), address(a), gemAmt);
+        vat.grab(ilk, a.proxy(), a.proxy(), address(this), dink, dart);
+
+        // make sure new position is as expected under the manager's ilk
+        (ink, art) = vat.urns(ilk, a.proxy());
+        assertEq(ink, 60 * 1e18);
+        assertEq(art, 41176470588235294117); // 40 * (1.05 / 1.02) * 1e18
+        assertEq(a.dai(), 0);
+        assertEq(a.gems(), 0);
+
+        // due to precision imperfection sin and vice are not zeroed back, but are negligible
+        assert(vat.sin(address(this)) < 1e30);
+        assert(vat.vice() < 1e30);
```

Comment:
The easiest way to "solve" this is by using `suck` and `heal` to transfer the `sin` dust to the Vow.

Reply:
`a != address(Vow) => Vat.sin(a) == 0` is an invariant we should strive to maintain for conceptual reasons, so I am in favor of doing what I suggested above.

Reply:
Actually the even easier solution is to use the Vow as the `w` argument to both `grab` calls--then the `sin` dust just ends up in the right place automatically, no extra work required.

Reply:
Done.